### PR TITLE
Trd 251 tradelogs allow register webhook on new trade

### PR DESCRIPTION
**Pull request by ngocthanh1389**

### CODE REVIEW

File: internal/server/tradelogs/broadcaster.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			if err := c.ws.WriteJSON(log); err != nil {
+				b.l.Errorw("error when send msg", "err", err)
+			}
+		}
+		b.mu.Unlock()
+	}
+}
+
+func (b *Broadcaster) addConn(event, maker string, conn *websocket.Conn) {
+	ID := xid.New().String()
```
Comment (thanhpp) [inline internal/server/tradelogs/broadcaster.go:50]:
```suggestion
	id  := xid.New().String()
```

File: internal/server/tradelogs/broadcaster.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+						b.l.Infow("socket is closed", "id", id)
+						delete(cons, id)
+					}
+				}
+			}
+		}
+		b.mu.Unlock()
+		time.Sleep(time.Minute)
+	}
+}
```
Comment (thanhpp) [inline internal/server/tradelogs/broadcaster.go:87]: ❓ what will be returned if there is no message

File: internal/server/tradelogs/broadcaster.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/rs/xid"
+	"go.uber.org/zap"
+)
+
+type Con struct {
+	id        string
+	ws        *websocket.Conn
+	eventHash string
+	maker     string
+}
```
Comment (thanhpp) [inline internal/server/tradelogs/broadcaster.go:19]: What if a connection want to subscribe to multiple eventHashes and makers?

**Review by Haiss2 — APPROVED** [review]
LGTM

File: internal/server/tradelogs/broadcaster.go (hunk 3 above)
Comment (ngocthanh1389) [reply internal/server/tradelogs/broadcaster.go:19]: we don't do that here

File: internal/server/tradelogs/broadcaster.go (hunk 2 above)
Comment (ngocthanh1389) [reply internal/server/tradelogs/broadcaster.go:87]: no msg, read message will return nil

File: internal/server/tradelogs/broadcaster.go (hunk 1 above)
Comment (ngocthanh1389) [reply internal/server/tradelogs/broadcaster.go]: fixed

**Pull request merged**